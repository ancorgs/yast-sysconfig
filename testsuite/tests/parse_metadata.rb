# encoding: utf-8

# Sysconfig module - testsuite
#
# parse_metadata tests
#
# testedfiles: Sysconfig.ycp
#
# $Id$
#
module Yast
  class ParseMetadataClient < Client
    def main
      Yast.import "Testsuite"
      Yast.import "Pkg" # override packamanager
      Yast.import "Sysconfig"

      Testsuite.Test(lambda { Sysconfig.parse_metadata(nil) }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { Sysconfig.parse_metadata("") }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { Sysconfig.parse_metadata("    ") }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { Sysconfig.parse_metadata("## Type: abc") }, [
        {},
        {},
        {}
      ], nil)
      Testsuite.Test(lambda do
        Sysconfig.parse_metadata("## Type: abc\n## Default: x")
      end, [
        {},
        {},
        {}
      ], nil)
      Testsuite.Test(lambda do
        Sysconfig.parse_metadata("## Type: abc\n\n##Default:x\n\n ")
      end, [
        {},
        {},
        {}
      ], nil)
      # multiline
      Testsuite.Test(lambda do
        Sysconfig.parse_metadata(
          "## Command: echo sdmvf,vm \\\n" +
            "## sdgfsdg sdf sf sf sdf \n" +
            "## Config:x, asdf\n" +
            "\n" +
            " "
        )
      end, [
        {},
        {},
        {}
      ], nil)

      nil
    end
  end
end

Yast::ParseMetadataClient.new.main
