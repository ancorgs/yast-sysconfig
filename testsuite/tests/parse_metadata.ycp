// Sysconfig module - testsuite
//
// parse_metadata tests
//
// $Id$
//

{

include "testsuite.ycp";

import "Sysconfig";

TEST( ``(Sysconfig::parse_metadata(nil) ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::parse_metadata("") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::parse_metadata("    ") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::parse_metadata("## Type: abc") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::parse_metadata("## Type: abc\n## Default: x") ), [ $[], $[], $[] ], nil );
TEST( ``(Sysconfig::parse_metadata("## Type: abc\n\n##Default:x\n\n ") ), [ $[], $[], $[] ], nil );

}

