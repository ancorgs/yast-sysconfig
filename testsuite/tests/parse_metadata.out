Return	$[]
Return	$[]
Return	$[]
Return	$["Type":"abc"]
Return	$["Default":"x", "Type":"abc"]
Return	$["Default":"x", "Type":"abc"]
